compiler/src/main/scala/org/scalajs/nscplugin/TypeConversions.scala

```diff
@@ -40,19 +40,19 @@ trait TypeConversions[G <: Global with Singleton] extends SubComponent {
     )
   }
 
-  private lazy val primitiveClassRefNameMap: Map[Symbol, String] = {
+  private lazy val primitiveRefMap: Map[Symbol, Types.NonArrayTypeRef] = {
     Map(
-        UnitClass    -> Definitions.VoidClass,
-        BooleanClass -> Definitions.BooleanClass,
-        CharClass    -> Definitions.CharClass,
-        ByteClass    -> Definitions.ByteClass,
-        ShortClass   -> Definitions.ShortClass,
-        IntClass     -> Definitions.IntClass,
-        LongClass    -> Definitions.LongClass,
-        FloatClass   -> Definitions.FloatClass,
-        DoubleClass  -> Definitions.DoubleClass,
-        NothingClass -> encodeClassFullName(RuntimeNothingClass),
-        NullClass    -> encodeClassFullName(RuntimeNullClass)
+        UnitClass    -> Types.VoidRef,
+        BooleanClass -> Types.BooleanRef,
+        CharClass    -> Types.CharRef,
+        ByteClass    -> Types.ByteRef,
+        ShortClass   -> Types.ShortRef,
+        IntClass     -> Types.IntRef,
+        LongClass    -> Types.LongRef,
+        FloatClass   -> Types.FloatRef,
+        DoubleClass  -> Types.DoubleRef,
+        NothingClass -> Types.ClassRef(encodeClassFullName(RuntimeNothingClass)),
+        NullClass    -> Types.ClassRef(encodeClassFullName(RuntimeNullClass))
```

Comment:
I know this is already here, but I don't understand why this is not simply the primitive type ref here...